packages/frontend/app/routes/tenants._index.tsx

```diff
@@ -0,0 +1,177 @@
+        const edgeNode = edge.node
+        if (edgeNode && sessionApiSecret === edgeNode.apiSecret) {
+          isOperator = edgeNode.isOperator
+          break
+        }
+      }
+    }
+    tenantEdges = isOperator
+      ? tenants.edges
+      : tenantEdges.filter(({ node }) => node.apiSecret === sessionApiSecret)
+  }
+  return json({
+    tenantEdges,
+    tenantPageInfo,
+    previousPageUrl,
+    nextPageUrl,
+    isOperator
+  })
+}
+
+export default function TenantsPage() {
+  const {
+    tenantEdges,
+    tenantPageInfo,
+    previousPageUrl,
+    nextPageUrl,
+    isOperator
+  } = useLoaderData<typeof loader>()
+  const navigate = useNavigate()
+
+  return (
+    <div className='pt-4 flex flex-col space-y-8'>
+      <div className='flex flex-col rounded-md bg-offwhite px-6'>
+        <PageHeader>
+          <div className='flex-1'>
+            <h3 className='text-2xl'>Tenants</h3>
+          </div>
+          <div className='ml-auto'>
+            {isOperator && (
+              <Button aria-label='add new tenant' to='/tenants/create'>
+                Add tenant
+              </Button>
+            )}
+          </div>
+        </PageHeader>
+        <Table>
+          <Table.Head columns={['Public name', 'Email', 'Status']} />
+          <Table.Body>
+            {tenantEdges.length ? (
+              tenantEdges.map((tenant) => (
+                <Table.Row
+                  key={tenant.node.id}
+                  className={tenant.node.deletedAt ? '' : 'cursor-pointer'}
+                  onClick={() =>
+                    tenant.node.deletedAt
+                      ? 'return'
+                      : navigate(`/tenants/${tenant.node.id}`)
+                  }
+                >
+                  <Table.Cell>
+                    <div className='flex flex-col'>
+                      {tenant.node.publicName ? (
+                        <span className='font-medium'>
+                          {tenant.node.publicName}{' '}
+                          {tenant.node.isOperator && (
+                            <span className='font-medium'> (Operator)</span>
+                          )}
+                        </span>
+                      ) : (
+                        <span className='text-tealish/80'>
+                          No public name{' '}
+                          {tenant.node.isOperator && (
+                            <span
+                              className='font-medium'
+                              title='This tenant is an operator tenant.'
+                            >
+                              {' '}
+                              (Operator)
+                            </span>
+                          )}
+                        </span>
+                      )}
+                      <div className='text-tealish/50 text-xs'>
+                        (ID: {tenant.node.id})
+                      </div>
+                    </div>
+                  </Table.Cell>
```

Comment:
![image](https://github.com/user-attachments/assets/ac01e34a-c6c7-431d-b052-1c5280580c8e)

If you like:

```suggestion
                  <Table.Cell>
                    <div className='flex flex-col'>
                      <div>
                        <span className='mr-2'>
                          {tenant.node.publicName ? (
                            <span className='font-medium'>
                              {tenant.node.publicName}
                            </span>
                          ) : (
                            <span className='text-tealish/80'>
                              No public name
                            </span>
                          )}
                        </span>
                        {tenant.node.isOperator && (
                          <Badge color={BadgeColor.Yellow}>Operator</Badge>
                        )}
                      </div>
                      <div className='text-tealish/50 text-xs'>
                        (ID: {tenant.node.id})
                      </div>
                    </div>
                  </Table.Cell>
```